WithSecure/withsecure/response_action.py

```diff
@@ -0,0 +1,50 @@
+from threading import Event
+from typing import Any
+
+from sekoia_automation.action import Action
+
+from withsecure.client import ApiClient
+from withsecure.constants import API_RESPONSE_ACTIONS_URL, API_TIMEOUT
+
+
+class ResponseAction(Action):
+    def run(self, arguments: Any) -> Any:
+        raise NotImplementedError()
+
+    def _execute_operation_on_device(
+        self, action_type: str, target: str, organization_id: str, parameters: dict[str, Any] | None = None
+    ) -> Any:
+        self.log(
+            f"Execute the action type '{action_type}' on device '{target}'",
+            level="debug",
+        )
+
+        request_payload: dict[str, Any] = {
+            "type": action_type,
+            "organizationId": organization_id,
+            "targets": [target],
+        }
+        if parameters:
+            # add only parameters with values
+            params_with_values = {k: v for k, v in parameters.items() if v is not None}
+            if len(params_with_values) > 0:
+                request_payload["parameters"] = params_with_values
+
+        headers = {"Accept": "application/json"}
+        # create the API client
+        client = ApiClient(
+            client_id=self.module.configuration.client_id,
+            secret=self.module.configuration.secret,
+            scope="connect.api.read connect.api.write",
+            stop_event=Event(),
+            log_cb=self.log,
+        )
+        response = client.post(
+            API_RESPONSE_ACTIONS_URL,
+            timeout=API_TIMEOUT,
+            json=request_payload,
+            headers=headers,
+        )
+        response.raise_for_status()
```

Comment:
I think we should extract the message in the failed response (maybe in the `run` method):
```json
{
  "message": "Bad request because of invalid parameters or data.",
  "code": 400000,
  "transactionId": "0000-d6e9bdb9f82946f3"
}
```